File: UI/rbike_mobile/lib/models/bike_favorite.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rbike_mobile/models/bike.dart';

part 'bike_favorite.g.dart';

@JsonSerializable()
class BikeFavorite {
  int? favoriteId;
  int? bikeId;
  int? userId;
  String? bikeName;
  String? username;
  Bike? bike;

  BikeFavorite();

  factory BikeFavorite.fromJson(Map<String, dynamic> json) {
    final bikeFavorite = _$BikeFavoriteFromJson(json);
    if (json['bike'] != null) {
      bikeFavorite.bike = Bike.fromJson(json['bike'] as Map<String, dynamic>);
      bikeFavorite.bikeName = bikeFavorite.bike!.name;
    }
    if (json['user'] != null) {
      bikeFavorite.username = json['user']['username'] as String?;
    }
    return bikeFavorite;
  }

  Map<String, dynamic> toJson() => _$BikeFavoriteToJson(this);
} 

File: rBike.Model/BikeFavorite.cs
using System;

namespace rBike.Model
{
    public class BikeFavorite
    {
        public int FavoriteId { get; set; }

        public int BikeId { get; set; }

        public int UserId { get; set; }

        public virtual Bike Bike { get; set; } = null!;

        public virtual User User { get; set; } = null!;

        public string? BikeName { get; set; }
        public string? Username { get; set; }
    }
}
